haskell-wrapper/src/RustBLS.hs

```diff
@@ -406,24 +388,18 @@ instance UnivariateFieldPolyVec Fr (RustPolyVec Fr) where
     where
       runFFT :: IO (RustPolyVec Fr size)
       runFFT = do
-          dl <- dlopen libPath [RTLD_NOW]
-          fftPtr <- dlsym dl "rust_wrapper_div_fft"
-          let !fft = mkFunDivFFT $ castFunPtr fftPtr
-
-          let valueSize = (fromInteger $ naturalToInteger $ value @size)
-          out <- callocForeignPtrBytes @CChar (scalarSize P.* (fromInteger $ naturalToInteger $ value @size))
-
-          withForeignPtr (rawData $ rawPoly l) $ \lPtr -> do
-            withForeignPtr (rawData $ rawPoly r) $ \rPtr -> do
-              withForeignPtr out $ \outPtr -> do
-                fft
-                  (castPtr lPtr) (valueSize P.* scalarSize)
-                  (castPtr rPtr) (valueSize P.* scalarSize)
-                  (valueSize P.* scalarSize) (castPtr outPtr)
+        let valueSize = (fromInteger $ naturalToInteger $ value @size)
+        out <- callocForeignPtrBytes @CChar (scalarSize P.* (fromInteger $ naturalToInteger $ value @size))
 
-          dlclose dl
+        withForeignPtr (rawData $ rawPoly l) $ \lPtr -> do
```

Comment:
This pattern with three nested `withForeignPtr` followed by `cast`s is used in many places. Maybe it can be moved to a separate function which will be called in these places?